crates/net/network/src/discovery.rs

```diff
@@ -43,23 +43,25 @@ impl Discovery {
     pub async fn new(
         discovery_addr: SocketAddr,
         sk: SecretKey,
-        dsicv4_config: Discv4Config,
+        discv4_config: Discv4Config,
     ) -> Result<Self, NetworkError> {
         let local_enr = NodeRecord::from_secret_key(discovery_addr, &sk);
         let (discv4, mut discv4_service) =
-            Discv4::bind(discovery_addr, local_enr, sk, dsicv4_config.clone())
+            Discv4::bind(discovery_addr, local_enr, sk, discv4_config.clone())
```

Comment:
we don't want to open the socket if discovery should be disabled, see comment above.

Reply:
Do you want the `Discovery` struct to be made even when Discv4 is disabled? Lighthouse makes their equivalent to maintain the nodes current ENR.

Reply:
that's a good point, so what we should do is, only spawn discovery if `disable_discovery` is `false`, but keep the `Discoveryconfig`, so the function should that the flag as argument

Reply:
I apologize for the long delay. I updated Discovery to have many of the fields as optional because they are directly related to the service itself. It might be cleaner to add those fields to the Discv4 struct instead of Discovery, but I'm not so sure.

Reply:
thanks, this seems fine.